app/controllers/tasks_controller.rb

```diff
@@ -1,29 +1,31 @@
 class TasksController < ApplicationController
+  before_action :check_for_user, only: [:create]
 
   def new
     @task = Task.new
     @task_categories = TaskCategory.all
   end
 
+
   def create
-    if current_user.nil?
-      session[:task] = task_params
-      redirect_to new_requester_session_path
+    @task = Task.new(task_params)
+    @task.user = current_user
+    @task.task_category_id = params[:task][:task_category_id]
+    if @task.save
+      redirect_to task_path(@task)
+      flash[:notice] = 'Task was successfully posted!'
     else
-      @task = Task.new(task_params)
-      @task.requester_id = current_user.id
-      @task.task_category_id = params[:task][:task_category_id]
-      if @task.save
-        redirect_to requesters_task_path(@task)
-        flash[:notice] = 'Task was successfully posted!'
-      else
-        render 'new'
-      end
+      render 'new'
     end
   end
 
   def index
-    @tasks = Task.where("status = ? OR status = ?", "Open", "Bidding")
+    if params[:type] == 'all'
+      @tasks = Task.where('status = ? OR status = ?', 'Open', 'Bidding')
```

Comment:
I took a look at this and from what I remember you can pass in several values to `where`. I would like you to try this: 
```ruby
@tasks = Task.where(status: ['Open', 'Bidding'})
```
Less SQL, more Rails ;-)